analysis/webservice/algorithms/doms/BaseDomsHandler.py

```diff
@@ -124,41 +127,63 @@ def __packValues(csv_mem_file, results):
 
         headers = [
             # Primary
-            "id", "source", "lon", "lat", "time", "platform", "sea_water_salinity_depth", "sea_water_salinity",
-            "sea_water_temperature_depth", "sea_water_temperature", "wind_speed", "wind_direction", "wind_u", "wind_v",
+            "id", "source", "lon (degrees_east)", "lat (degrees_north)", "time", "platform",
+            "sea_surface_salinity (1e-3)", "sea_surface_temperature (degree_C)", "wind_speed (m s-1)", "wind_direction",
+            "wind_u (m s-1)", "wind_v (m s-1)",
             # Match
-            "id", "source", "lon", "lat", "time", "platform", "sea_water_salinity_depth", "sea_water_salinity",
-            "sea_water_temperature_depth", "sea_water_temperature", "wind_speed", "wind_direction", "wind_u", "wind_v"
+            "id", "source", "lon (degrees_east)", "lat (degrees_north)", "time", "platform",
+            "depth (m)", "sea_water_salinity (1e-3)",
+            "sea_water_temperature (degree_C)", "wind_speed (m s-1)",
+            "wind_direction", "wind_u (m s-1)", "wind_v (m s-1)"
         ]
 
         writer.writerow(headers)
 
+        #
+        # Compress multiple depth variables into one depth array
+        #
+        depthAggregate = []
```

Comment:
I'm a little confused about this depth logic, can you explain? It seems like you are picking either the SSS or SST depth if one exists. Are you sure that in all cases depth will be the same between SSS and SST insitu measurements? Also, is it necessary to iterate the matches twice; couldn't this if statement be done as part of the row construction below?

Reply:
I think this needs further discussion. I know ICOADS stores two separate depth fields (one for sst and one for sss) per record in their index. I'm not sure about FSU but this is why we agreed to have two fields for depth in edge. I think by trying to combine it into one value we are making a decision for the in situ provider when we should just be passing along the data that was given to us. 

Same for filling missing data with 0; that implies that the in situ provider said that the data was recorded with a depth of 0. For example the FSU translation specification states "Note that a SAMOS value of -9999 or -8888 is interpreted as “missing” or “special” and is left out of DOMS." for Sea_water_temperature_depth. In this case we might end up reporting a 0 depth when that is in fact incorrect.

IMO we need to retain two depth fields: Sea_water_temperature_depth and Sea_water_salinity_depth